prime-router/src/test/kotlin/fhirengine/translation/hl7/FhirTransformerTests.kt

```diff
@@ -763,4 +769,36 @@ class FhirTransformerTests {
         assertThat(resource.meta.lastUpdated).isEqualTo(bundle.timestamp)
         assertThat(newValue[0].primitiveValue()).isEqualTo("2021-08-09T05:52:00.000-07:00")
     }
+
+    @Test
+    fun `test extending schema overwrite element`() {
+        val actionLogger = ActionLogger()
+        val fhirBundle = File("src/test/resources/fhirengine/engine/valid_data.fhir").readText()
+        val messages = FhirTranscoder.getBundles(fhirBundle, actionLogger)
+        val bundle = messages[0]
+
+        val childSchema = ConfigSchemaReader.readSchemaTreeFromFile(
+            "test_extension_schema",
+            "src/test/resources/fhir_sender_transforms",
+            schemaClass = FhirTransformSchema::class.java,
+        ) as FhirTransformSchema
+
+        val transformer = FhirTransformer(childSchema)
+        val transformedBundle = transformer.transform(bundle)
+
+        val transformedDiagnosticReports = mutableListOf<DiagnosticReport>()
+        var transformedPatient = Patient()
+        transformedBundle.entry.forEach {
+            when (val resource = it.resource) {
+                is Patient -> transformedPatient = resource
+                is DiagnosticReport -> transformedDiagnosticReports.add(resource)
+            }
+        }
+
+        val transformedObservation = transformedDiagnosticReports[0].result[0].resource as Observation
+
+        assertThat(transformedDiagnosticReports[0].id).isEqualTo("extensionId")
+        assertThat(transformedPatient.id).isEqualTo("123456")
+        assertThat(transformedObservation.status).isEqualTo(Observation.ObservationStatus.FINAL)
```

Comment:
I think the test needs to also verify that extending a schema uses elements from the base schema. Since there's only one element in the base schema and that element is being overridden, we can't really verify that here, so adding another element to the base schema and testing that would be helpful.